packages/ag-charts-website/src/pages/gallery/examples/[exampleName]/[theme]-plain.[format].ts

```diff
@@ -13,18 +15,61 @@ interface Params {
     format: (typeof formats)[number];
 }
 
+interface Props {
+    uncompressed?: Buffer;
+}
+
+interface Page {
+    params: Params;
+    props: Props;
+}
+
 export async function getStaticPaths() {
     const galleryDataEntry = await getEntry('gallery', 'data');
     const pages = getGalleryExampleThemePages({ galleryData: galleryDataEntry.data });
-    const result: { params: Params }[] = [];
-    for (const page of pages) {
-        formats.forEach((format) => {
-            result.push({ params: { ...page.params, format } });
+
+    const result: Page[] = [];
+    for (let i = 0; i < pages.length; i += THUMBNAIL_POOL_SIZE) {
+        const chunk = pages.slice(i, i + THUMBNAIL_POOL_SIZE);
+
+        const examples = await Promise.all(
+            chunk.map(async ({ params }): Promise<Buffer> => {
+                const { exampleName, theme } = params;
+                const uncompressed = await generateExample({ exampleName, theme });
+                return uncompressed;
+            })
+        );
+        const subPages = chunk.flatMap(({ params }, index): Page[] => {
+            const { exampleName, theme } = params;
+            const uncompressed = examples[index];
+            return formats.map((format) => ({
+                params: { exampleName, theme, format },
+                props: { uncompressed },
```

Comment:
I'm a bit concerned with the memory overhead this might imply - can we come back to this once I've had time to dig into a Nx-based task decomposition for example generation, which may include thumbnail generation coming from a completely different approach?